### PR TITLE
docs: add mui tutorial and example

### PR DESCRIPTION
**Pull request by biskuvit**
I am going to create a material-ui tutorial.

- [X] Example
- [X] Documentation
- [x] Documentation images

https://codesandbox.io/embed/github/pankod/refine/tree/next/examples/tutorial/mui

### CODE REVIEW

File: documentation/docs/ui-frameworks/mui/tutorial.md [hunk 1]
```diff
@@ -0,0 +1,1745 @@
...
+
+<br/>
+
+Let's break down the `<PostList/>` component to understand what's going on here:
+
+✳️ `<DataGrid/>` is a native **Material UI** component. It renders records row by row as a table.
+`<DataGrid/>` expects a `columns` prop as a required.
+
+:::note
+**refine** uses [Material UI](https://mui.com/x/react-data-grid/#mit-version) components to render data.
```
Comment (omeraplak) [inline documentation/docs/ui-frameworks/mui/tutorial.md:477]: here we can specify that we also support the `<DataGridPro />` component

File: documentation/docs/ui-frameworks/mui/tutorial.md [hunk 2]
```diff
@@ -0,0 +1,1745 @@
...
+        </List>
+    );
+};
+```
+
+We construct an array of `categoryId`'s from `/posts` endpoint and pass it to the `useOne` hook. `categoriesData` will be filled with _id-title_ tuples to be used for rendering our component.
+
+Try the result on your browser and you'll notice that the category column is filled correctly with the matching category titles for the each record's category id's. Even the loading state is managed by **refine**.
+
+:::
```
Comment (omeraplak) [inline documentation/docs/ui-frameworks/mui/tutorial.md:705]:
```suggestion
```

**Review by omeraplak — CHANGES_REQUESTED** [review]

File: .changeset/kind-cameras-lick.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+---
+"mui-tutorial": patch
+---
+
+create refine [material ui tutorial](https://refine.dev/docs/ui-frameworks/mui/tutorial/) with [example](https://github.com/pankod/refine/tree/master/examples/tutorial/mui)
```
Comment (yildirayunlu) [inline .changeset/kind-cameras-lick.md:5]: For example, don't add changeset.

**Review by yildirayunlu — CHANGES_REQUESTED** [review]

File: documentation/docs/ui-frameworks/mui/tutorial.md [hunk 4]
```diff
@@ -0,0 +1,1705 @@
...
+        >
+            <Box
+                component="form"
+                sx={{ display: "flex", flexDirection: "column" }}
+                autoComplete="off"
+            >
+                <TextField
+                    {...register("title", { required: true })}
+                    error={!!errors?.title}
+                    helperText={errors?.title?.message}
```
Comment (salihozdemir) [inline documentation/docs/ui-frameworks/mui/tutorial.md:1058]:
```suggestion
                    {...register("title", { required: "This field is required" })}
                    error={!!errors.title}
                    helperText={errors.title?.message}
```

File: documentation/docs/ui-frameworks/mui/tutorial.md [hunk 5]
```diff
@@ -0,0 +1,1705 @@
...
+                    margin="normal"
+                    required
+                    fullWidth
+                    id="title"
+                    label="Title"
+                    name="title"
+                    defaultValue={" "}
```
Comment (salihozdemir) [inline documentation/docs/ui-frameworks/mui/tutorial.md:1065]:
```suggestion
                   
```

Is it necessary?

File: documentation/docs/ui-frameworks/mui/tutorial.md [hunk 6]
```diff
@@ -0,0 +1,1705 @@
...
+                                    error={!!errors.status}
+                                    helperText={
+                                        errors.status && "status required"
+                                    }
+                                    required
+                                />
+                            )}
+                        />
+                    )}
+                />
```
Comment (salihozdemir) [inline documentation/docs/ui-frameworks/mui/tutorial.md:1094]:
```suggestion
                <Controller
                    control={control}
                    name="status"
                    rules={{ required: "This field is required" }}
                    render={({ field }) => (
                        <Autocomplete
                            {...field}
                            options={["published", "draft", "rejected"]}
                            onChange={(_, value) => {
                                field.onChange(value);
                            }}
                            renderInput={(params) => (
                                <TextField
                                    {...params}
                                    label="status"
                                    margin="normal"
                                    variant="outlined"
                                    error={!!errors.status}
                                    helperText={errors.status?.message}
                                    required
                                />
                            )}
                        />
                    )}
                />
```

File: documentation/docs/ui-frameworks/mui/tutorial.md [hunk 7]
```diff
@@ -0,0 +1,1705 @@
...
+                <Controller
+                    control={control}
+                    name="category"
+                    rules={{ required: true }}
+                    defaultValue=""
```
Comment (salihozdemir) [inline documentation/docs/ui-frameworks/mui/tutorial.md:1099]:
```suggestion
                    rules={{ required: "This field is required" }}
```

File: documentation/docs/ui-frameworks/mui/tutorial.md [hunk 8]
```diff
@@ -0,0 +1,1705 @@
...
+                            renderInput={(params) => (
+                                <TextField
+                                    {...params}
+                                    label="Category"
+                                    margin="normal"
+                                    variant="outlined"
+                                    error={!!errors.category}
+                                    helperText={
+                                        errors.category && "category required"
+                                    }
```
Comment (salihozdemir) [inline documentation/docs/ui-frameworks/mui/tutorial.md:1129]:
```suggestion
                                    helperText={errors.category?.message}
```

File: documentation/docs/ui-frameworks/mui/tutorial.md [hunk 9]
```diff
@@ -0,0 +1,1705 @@
...
+        <List>
+            <DataGrid
+                {...restDataGridProps}
+                filterMode={filterMode}
+                filterModel={filterModel}
+                onFilterModelChange={onFilterModelChange}
+                autoHeight
+            />
+        </List>
+    );
```
Comment (salihozdemir) [inline documentation/docs/ui-frameworks/mui/tutorial.md:997]:
```suggestion 
    return (
        <List>
            <DataGrid {...dataGridProps} autoHeight />
        </List>
    );
```

File: documentation/docs/ui-frameworks/antd/tutorial.md [hunk 10]
```diff
@@ -1215,7 +1226,7 @@ In edit page, `useForm` hook initializes the form with current record values.
 
 ✳️ Form data is set automatically, whenever children inputs `<Form.Item>`'s are edited.
 
-✳️ Save button submits the form by executing the `useUpdate` method provided by the [`dataProvider`](/core/providers/data-provider.md). After a succesfull response, the application will be redirected to the listing page.
+✳️ Save button submits the form by executing the `useUpdate` method provided by the [`dataProvider`](/core/providers/data-provider.md). After a successfull response, the application will be redirected to the listing page.
```
Comment (ozkalai) [inline documentation/docs/ui-frameworks/antd/tutorial.md:1229]:
```suggestion
✳️ Save button submits the form by executing the `useUpdate` method provided by the [`dataProvider`](/core/providers/data-provider.md). After a successful response, the application will be redirected to the listing page.
```

File: examples/tutorial/mui/src/pages/posts/list.tsx [hunk 11]
```diff
@@ -0,0 +1,99 @@
...
+    Stack,
+} from "@pankod/refine-mui";
+
+import { IPost, ICategory } from "interfaces";
+
+export const PostList: React.FC = () => {
+    const columns: GridColumns = [
+        { field: "title", headerName: "Title", flex: 1, minWidth: 350 },
+        {
+            field: "category.id",
```
Comment (ozkalai) [inline examples/tutorial/mui/src/pages/posts/list.tsx:21]: you might have to add flex key for wider screen.

File: examples/tutorial/mui/src/pages/posts/list.tsx [hunk 12]
```diff
@@ -0,0 +1,99 @@
...
+                    resource: "categories",
+                    id: params.row.category.id,
+                });
+                return data?.data.title;
+            },
+        },
+        {
+            field: "status",
+            headerName: "Status",
+            minWidth: 150,
```
Comment (ozkalai) [inline examples/tutorial/mui/src/pages/posts/list.tsx:36]: you might have to add flex key for wider screen.

File: examples/tutorial/mui/src/pages/posts/list.tsx [hunk 13]
```diff
@@ -0,0 +1,99 @@
...
+            renderCell: function render(params) {
+                return <TagField value={params.row.status} />;
+            },
+        },
+        {
+            field: "createdAt",
+            headerName: "CreatedAt",
+            minWidth: 220,
```
Comment (ozkalai) [inline examples/tutorial/mui/src/pages/posts/list.tsx:44]: you might have to add flex key for wider screen.

File: examples/tutorial/mui/src/pages/posts/list.tsx [hunk 14]
```diff
@@ -0,0 +1,99 @@
...
+            renderCell: function render(params) {
+                return <DateField format="LLL" value={params.row.createdAt} />;
+            },
+        },
+        {
+            headerName: "Actions",
+            field: "actions",
+            minWidth: 250,
```
Comment (ozkalai) [inline examples/tutorial/mui/src/pages/posts/list.tsx:52]: you might have to add flex key for wider screen.

File: documentation/docs/ui-frameworks/mui/tutorial.md (hunk 5 above)
Comment (biskuvit) [reply documentation/docs/ui-frameworks/mui/tutorial.md:1065]: Actually, It's required to get the default value

File: examples/tutorial/mui/src/pages/posts/edit.tsx [hunk 15]
```diff
@@ -0,0 +1,121 @@
...
+    const { autocompleteProps } = useAutocomplete<ICategory>({
+        resource: "categories",
+        defaultValue: queryResult?.data?.data.category.id,
+        queryOptions: { enabled: !!queryResult?.data?.data.category.id },
+    });
+
+    return (
+        <Edit
+            isLoading={formLoading}
+            saveButtonProps={{ onClick: handleSubmit(onFinish) }}
```
Comment (aliemir) [inline examples/tutorial/mui/src/pages/posts/edit.tsx:30]:
```suggestion
        saveButtonProps,
        control,
        formState: { errors },
    } = useForm();

    const { autocompleteProps } = useAutocomplete<ICategory>({
        resource: "categories",
        defaultValue: queryResult?.data?.data.category.id,
        queryOptions: { enabled: !!queryResult?.data?.data.category.id },
    });

    return (
        <Edit
            isLoading={formLoading}
            saveButtonProps={saveButtonProps}
```

**Review by omeraplak — APPROVED** [review]

File: documentation/docs/ui-frameworks/mui/tutorial.md [hunk 16]
```diff
@@ -0,0 +1,1740 @@
...
+};
+```
+
+After adding `canDelete` prop, `<DeleteButton>` will appear in edit form.
+
+## Live Codesandbox Example
+
+Our tutorial is complete. Below you'll find a live Codesandbox example displaying what we have done so far:
+
+<iframe src="https://codesandbox.io/embed/github/pankod/refine/tree/next/examples/tutorial/mui?autoresize=1&fontsize=14&theme=dark&view=preview"
```
Comment (omeraplak) [inline documentation/docs/ui-frameworks/mui/tutorial.md:1723]:
```suggestion
<iframe src="https://codesandbox.io/embed/github/pankod/refine/tree/master/examples/tutorial/mui?autoresize=1&fontsize=14&theme=dark&view=preview"
```

**Review by omeraplak — CHANGES_REQUESTED** [review]

File: examples/tutorial/mui/src/index.tsx [hunk 17]
```diff
@@ -0,0 +1,30 @@
...
+    }
+    &.SnackbarItem-variantSuccess {
+        background-color: ${(props) => props.theme.palette.primary.main};
+        color: ${(props) => props.theme.palette.primary.contrastText};
+    }
+    &.SnackbarItem-variantError {
+        background-color: ${(props) => props.theme.palette.error.main};
+        color: ${(props) => props.theme.palette.common.white};
+    }
+`;
```
Comment (salihozdemir) [inline examples/tutorial/mui/src/index.tsx:21]: I think, we should move it to the `App.tsx`

File: documentation/docs/ui-frameworks/mui/tutorial.md [hunk 18]
```diff
@@ -0,0 +1,1740 @@
...
+        <ThemeProvider theme={LightTheme}>
+            <CssBaseline />
+            <GlobalStyles styles={{ html: { WebkitFontSmoothing: "auto" } }} />
+            <Refine
+                routerProvider={routerProvider}
+                dataProvider={dataProvider("https://api.fake-rest.refine.dev")}
+                Layout={Layout}
+                ReadyPage={ReadyPage}
+                catchAll={<ErrorComponent />}
+            />
```
Comment (salihozdemir) [inline documentation/docs/ui-frameworks/mui/tutorial.md:219]: Should we add the `notificationProvider`?

File: documentation/docs/ui-frameworks/mui/tutorial.md [hunk 19]
```diff
@@ -0,0 +1,1740 @@
...
+        </List>
+    );
+};
+```
+
+We access the row record with `valueGetter` and pass categoryId to the `useOne` hook. `useOne` hook returns a `data` object which contains the category title.
+
+Try the result on your browser and you'll notice that the category column is filled correctly with the matching category titles for the each record's category id's. Even the loading state is managed by **refine**.
+
+To get more detailed information about this hook, please refer the [useOne Documentation](#).
```
Comment (salihozdemir) [inline documentation/docs/ui-frameworks/mui/tutorial.md:708]:
```suggestion
To get more detailed information about this hook, please refer the [useOne Documentation](#).
```
Missing link

**Review by salihozdemir — APPROVED** [review]

File: documentation/docs/ui-frameworks/mui/tutorial.md [hunk 20]
```diff
@@ -0,0 +1,1609 @@
...
+To get more detailed information about this hook, please refer the [useOne Documentation](/core/hooks/data/useOne.md).
+
+## Adding search and filters
+
+We're done with displaying `post` records on our `<DataGrid>`. Let's see search and filtering capabilities to the component, so that the user can have more control over the data.
+
+`useDataGrid` hook provides a native `sorting` and `filters` props. Our way to handle these is to use the `useDataGrid` hook and pass the props to the `<DataGrid>` component.
+
+:::tip
+To see how the filtering works and more detail, you can look at the [`useDataGrid`](/ui-frameworks/mui/hooks/useDataGrid.md) hook.
```
Comment (omeraplak) [inline documentation/docs/ui-frameworks/mui/tutorial.md:717]:
```suggestion
To see how the filtering works and more detail, you can look at the [`useDataGrid`](/ui-frameworks/mui/hooks/useDataGrid.md#sorting) hook.
```

Comment (omeraplak) [reply documentation/docs/ui-frameworks/mui/tutorial.md:717]: I'm not sure about this syntax but my goal is to post it in the sorting section

**Review by aliemir — APPROVED** [review]

**Pull request merged**